The content of this file is derived from original patches published to libc-alpha at sourceware dot org
by Szabolcs Nagy. The original, unmodified, version is available in the patches/debian/2.27 folder of
this repository or from  https://sourceware.org/bugzilla/show_bug.cgi?id=19329#c9

The original content has been adapted by MathWorks to enable patching glibc 2.28 since there were 
source code changes in glibc since the original patches were constructed

NOTE: ADAPTED TO PATCH v2.28 via rpmbuild

---
 elf/dl-tls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/dl-tls.c b/elf/dl-tls.c
index d1866af1..46d1e1b8 100644
--- a/elf/dl-tls.c
+++ b/elf/dl-tls.c
@@ -643,7 +643,7 @@ _dl_allocate_tls_init (void *result)
 	}
 
       total += cnt;
-      if (total >= dtv_slots)
+      if (total > dtv_slots)
 	break;
 
       /* Synchronize with dl_add_to_slotinfo.  */
-- 
2.27.0

